lightly/transforms/byol_transform.py

```diff
@@ -0,0 +1,172 @@
+            T.RandomHorizontalFlip(p=hf_prob),
+            T.RandomVerticalFlip(p=vf_prob),
+            T.RandomApply([color_jitter], p=cj_prob),
+            T.RandomGrayscale(p=random_gray_scale),
+            GaussianBlur(kernel_size=kernel_size, sigmas=sigmas, prob=gaussian_blur),
+            RandomSolarization(prob=solarization_prob),
+            T.ToTensor(),
+        ]
+        if normalize:
+            transform += [T.Normalize(mean=normalize["mean"], std=normalize["std"])]
+        self.transform = T.Compose(transform)
+
+    def __call__(self, image: Union[Tensor, Image]) -> Tensor:
+        """
+        Applies the transforms to the input image.
+
+        Args:
+            image:
+                The input image to apply the transforms to.
+
+        Returns:
+            The transformed image.
+
+        """
+        return self.transform(image)
+
+
+class BYOLView2Transform:
+    def __init__(
+        self,
+        input_size: int = 224,
+        cj_prob: float = 0.8,
+        cj_strength: float = 1.0,
+        cj_bright: float = 0.4,
+        cj_contrast: float = 0.4,
+        cj_sat: float = 0.2,
+        cj_hue: float = 0.1,
+        min_scale: float = 0.08,
+        random_gray_scale: float = 0.2,
+        gaussian_blur: float = 0.1,
+        solarization_prob: float = 0.2,
+        kernel_size: Optional[float] = None,
+        sigmas: Tuple[float, float] = (0.1, 2),
+        vf_prob: float = 0.0,
+        hf_prob: float = 0.5,
+        rr_prob: float = 0.0,
+        rr_degrees: Union[None, float, Tuple[float, float]] = None,
+        normalize: Union[None, dict] = IMAGENET_NORMALIZE,
+    ):
+        color_jitter = T.ColorJitter(
+            brightness=cj_strength * cj_bright,
+            contrast=cj_strength * cj_contrast,
+            saturation=cj_strength * cj_sat,
+            hue=cj_strength * cj_hue,
+        )
+
+        transform = [
+            T.RandomResizedCrop(size=input_size, scale=(min_scale, 1.0)),
+            random_rotation_transform(rr_prob=rr_prob, rr_degrees=rr_degrees),
+            T.RandomHorizontalFlip(p=hf_prob),
+            T.RandomVerticalFlip(p=vf_prob),
+            T.RandomApply([color_jitter], p=cj_prob),
+            T.RandomGrayscale(p=random_gray_scale),
+            GaussianBlur(kernel_size=kernel_size, sigmas=sigmas, prob=gaussian_blur),
+            RandomSolarization(prob=solarization_prob),
+            T.ToTensor(),
+        ]
+        if normalize:
+            transform += [T.Normalize(mean=normalize["mean"], std=normalize["std"])]
+        self.transform = T.Compose(transform)
+
+    def __call__(self, image: Union[Tensor, Image]) -> Tensor:
+        """
+        Applies the transforms to the input image.
+
+        Args:
+            image:
+                The input image to apply the transforms to.
+
+        Returns:
+            The transformed image.
+
+        """
+        return self.transform(image)
+
+
+class BYOLTransform(MultiViewTransform):
```

Comment:
We put the main transform at the top of the file for the other transforms.

Reply:
I'm not sure whether I can switch the order as we use the `BYOLView1Transform` in the type hint and it has to be initialized before using it.